src/SonarQube.Client/Services/SonarQubeService.cs

```diff
@@ -0,0 +1,253 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarQube.Client.Models;
+using SonarQube.Client.Services;
+
+namespace SonarQube.Client
+{
+    public class SonarQubeService
+    {
+        internal const int MaximumPageSize = 500;
+        internal readonly Version OrganizationsFeatureMinimalVersion = new Version(6, 2);
+
+        private readonly ISonarQubeClient sonarqubeClient;
+
+        private ConnectionDTO connection;
+        private bool isConnected;
+        private Version serverVersion;
+
+        public SonarQubeService(ISonarQubeClient sonarqubeClient)
+        {
+            this.sonarqubeClient = sonarqubeClient;
+        }
+
+        public bool HasOrganizationsFeature
+        {
+            get
+            {
+                if (!this.isConnected)
+                {
+                    throw new InvalidOperationException("This operation expects the service to be connected.");
+                }
+
+                return this.serverVersion >= OrganizationsFeatureMinimalVersion;
+            }
+        }
+
+        public async Task ConnectAsync(Connection connection, CancellationToken token)
+        {
+            if (this.isConnected)
+            {
+                throw new InvalidOperationException("This operation expects the service to be connected.");
```

Comment:
should say the opposite - this operation expects the service to be not connected